addon/data/viewerworker.js

```diff
@@ -27,6 +27,51 @@ document.addEventListener("error", function (event) {
   self.port.emit("alertError", event.detail);
 }, false);
 
+document.addEventListener("request-account", function (event) {
+  self.port.emit("requestAccount", event.detail);
+});
+
+document.addEventListener("request-profile", function (event) {
+  self.port.emit("requestProfile", event.detail);
+}, false);
+
+document.addEventListener("request-profile-update", function (event) {
+  self.port.emit("requestProfileUpdate", event.detail);
+}, false);
+
+self.port.on("account", function (id, response) {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("got-account", true, true, JSON.stringify({
+    response,
+    id
+  }));
+  document.dispatchEvent(event);
+});
+
+self.port.on("profile", function (id, response) {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("got-profile", true, true, JSON.stringify({
+    response,
+    id
+  }));
+  document.dispatchEvent(event);
+});
+
+self.port.on("profileUpdate", function (id, response) {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("got-profile-update", true, true, JSON.stringify({
+    response,
+    id
+  }));
+  document.dispatchEvent(event);
+});
+
+self.port.on("profileRefresh", function () {
+  var event = document.createEvent("CustomEvent");
+  event.initCustomEvent("profile-refresh", true, true, null);
+  document.dispatchEvent(event);
+});
+
```

Comment:
Maybe all this messaging could be simpler.  We basically want to keep data in sync between the page and the addon.  I think we can consider the addon the canonical source?  So we have messages from the page asking to do things: to sign in, to create an account, to set the display name, and to set the avatar.  From the addon we have: set the state.  That includes the login status, profile information, display name, etc.  The addon may have to wait to set the state if the page isn't ready to accept the message yet, though I think the way the helper initializes, all the Javascript on the page should have executed (including event listeners) before the worker starts up.

So it seems to me we should have some events from the page: request-signin, request-create, set-profile-info

Then from the addon to the page: set-profile-info.

I've used a pattern of names-with-dashes in event names, and camelCase in message names.  This is a silly.  For the most part viewerworker just pipes stuff through, we should probably keep the exact same names, and potentially do some of these in a loop.